### DISCUSSION TITLE
Visualize navigation patterns from mission module

### DISCUSSION THREAD

**Discussion by Dennis-Wijngaarden** [paparazzi/paparazzi#3243] [Ideas]
Would it be possible to extend the mission module and GCS in such a way that the navigation patterns / waypoints defined in a active mission of the mission module can be visualized on the paparazzi GCS?

**Comment by gautierhattenberger** [comment]
This would be great, indeed, however it is not so easy. The main reason is that unless the GCS itself creates the mission elements, it won't be able to associate an idea with a specific pattern.
One way to overcome this would be listen to the mission messages to have an idea of the association between ID and patterns. This could be used for "standard" patterns, but not for the custom mission elements.
Note that at least, the segments and circles are displayed even when in mission mode.

**Comment by noether** [comment]
I do not know whether you refer to something like the following. We have integrated our GVF into the current GCS. Maybe you might be interested in looking at it to implement your suggestion?

https://paparazzi-uav.readthedocs.io/en/stable/user_guide/gvf_viewer.html
https://github.com/paparazzi/PprzGCS/pull/15

**Reply by Dennis-Wijngaarden** [reply]
Drawing shapes from the AC to the GCS might be a good option. Currently I'm updating the mavlink protocol in paparazzi to be able to upload and download mavlink missions to and from the drone. Those missions will be transferred to the existing mission module. I can visualize an active mavlink mission if I'm able to draw shapes on the GCS from the AC.

**Reply by gautierhattenberger** [reply]
You can already draw basic shapes using standard paparazzi messages. I guess you could also do that internally from a dedicated widget.
You can find some information about the GCS here: https://docs.paparazziuav.org/PprzGCS/index.html
